cmd/subnetcmd/upgradecmd/vm.go

```diff
@@ -0,0 +1,305 @@
+	case useLatest:
+		return updateToLatestVersion(subnetName, vmType, sc, networkToUpgrade)
+	case targetVersion != "":
+		return updateToSpecificVersion(subnetName, vmType, sc, networkToUpgrade)
+	case useBinary != "":
+		return updateToCustomBin(subnetName, vmType, sc, networkToUpgrade)
+	}
+
+	latestVersionUpdate := "Update to latest version"
+	specificVersionUpdate := "Update to a specific version"
+	customBinaryUpdate := "Update to a custom binary (coming soon)"
+
+	updateOptions := []string{latestVersionUpdate, specificVersionUpdate, customBinaryUpdate}
+
+	updatePrompt := "How would you like to update your subnet's virtual machine"
+	updateDecision, err := app.Prompt.CaptureList(updatePrompt, updateOptions)
+	if err != nil {
+		return err
+	}
+
+	switch updateDecision {
+	case latestVersionUpdate:
+		return updateToLatestVersion(subnetName, vmType, sc, networkToUpgrade)
+	case specificVersionUpdate:
+		return updateToSpecificVersion(subnetName, vmType, sc, networkToUpgrade)
+	case customBinaryUpdate:
+		return updateToCustomBin(subnetName, vmType, sc, networkToUpgrade)
+	default:
+		return errors.New("invalid option")
+	}
+}
+
+func updateToLatestVersion(subnetName string, vmType models.VMType, sc models.Sidecar, networkToUpgrade string) error {
+	// pull in current version
+	currentVersion := sc.VMVersion
+
+	// check latest version
+	latestVersion, err := binutils.GetLatestReleaseVersion(binutils.GetGithubLatestReleaseURL(
+		constants.AvaLabsOrg,
+		vmType.RepoName(),
+	))
+	if err != nil {
+		return err
+	}
+
+	// check if current version equals latest
+	if currentVersion == "latest" || currentVersion == latestVersion {
+		ux.Logger.PrintToUser("VM already up-to-date")
+		return nil
+	}
+
+	return updateVMByNetwork(sc, latestVersion, networkToUpgrade)
+}
+
+func updateToSpecificVersion(subnetName string, vmType models.VMType, sc models.Sidecar, networkToUpgrade string) error {
+	// pull in current version
+	currentVersion := sc.VMVersion
+
+	// Get version to update to
+	var err error
+	if targetVersion == "" {
+		targetVersion, err = app.Prompt.CaptureVersion("Enter version")
+		if err != nil {
+			return err
+		}
+	}
+
+	// check if current version equals chosen version
+	if currentVersion == targetVersion {
+		ux.Logger.PrintToUser("VM already up-to-date")
+		return nil
+	}
+
+	return updateVMByNetwork(sc, targetVersion, networkToUpgrade)
+}
+
+func updateVMByNetwork(sc models.Sidecar, targetVersion string, networkToUpgrade string) error {
+	switch networkToUpgrade {
+	case futureDeployment:
+		return updateFutureVM(sc, targetVersion)
+	case localDeployment:
+		return updateExistingLocalVM(sc, targetVersion)
+	case fujiDeployment:
+		return chooseManualOrAutomatic(sc, targetVersion, networkToUpgrade)
+	case mainnetDeployment:
+		return updateMainnetVM()
+	default:
+		return errors.New("unknown deployment")
+	}
+}
+
+func updateToCustomBin(subnetName string, vmType models.VMType, sc models.Sidecar, networkToUpgrade string) error {
```

Comment:
ups, needs TODO at least, and maybe a issue

Reply:
#319